ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts/uplinks.py

```diff
@@ -184,148 +184,239 @@ def uplinks(self: EosDesignsFacts) -> list:
 
-                # Remove vlans if upstream switch does not have them #}
-                if self.shared_utils.enable_trunk_groups:
-                    uplink["trunk_groups"] = ["UPLINK"]
-                    if self.shared_utils.mlag is True:
-                        uplink["peer_trunk_groups"] = [self.shared_utils.group]
-                    else:
-                        uplink["peer_trunk_groups"] = [self.shared_utils.hostname]
+            if self.shared_utils.configure_inband_mgmt:
+                # Always add inband_mgmt_vlan even if the uplink switch does not have this vlan defined
+                uplink_vlans.add(self.shared_utils.inband_mgmt_vlan)
 
-                uplink_vlans = set(self._vlans)
-                uplink_vlans = uplink_vlans.intersection(uplink_switch_facts._vlans)
+            uplink["vlans"] = list_compress(list(uplink_vlans)) if uplink_vlans else "none"
 
-                if self.shared_utils.configure_inband_mgmt:
-                    # Always add inband_mgmt_vlan even if the uplink switch does not have this vlan defined
-                    uplink_vlans.add(self.shared_utils.inband_mgmt_vlan)
+            if uplink_native_vlan := get(self.shared_utils.switch_data_combined, "uplink_native_vlan"):
+                uplink["native_vlan"] = uplink_native_vlan
 
-                if uplink_vlans:
-                    uplink["vlans"] = list_compress(list(uplink_vlans))
-                else:
-                    uplink["vlans"] = "none"
+            if self._short_esi is not None:
+                uplink["peer_short_esi"] = self._short_esi
 
-                if uplink_native_vlan := get(self.shared_utils.switch_data_combined, "uplink_native_vlan"):
-                    uplink["native_vlan"] = uplink_native_vlan
+            if self.shared_utils.link_tracking_groups is not None:
+                uplink["link_tracking_groups"] = [{"name": lt_group["name"], "direction": "upstream"} for lt_group in self.shared_utils.link_tracking_groups]
+            if self.shared_utils.uplink_structured_config is not None:
+                uplink["structured_config"] = self.shared_utils.uplink_structured_config
 
-                if self._short_esi is not None:
-                    uplink["peer_short_esi"] = self._short_esi
+            uplinks.append(uplink)
+        return uplinks
 
-                if self.shared_utils.link_tracking_groups is not None:
-                    uplink["link_tracking_groups"] = []
-                    for lt_group in self.shared_utils.link_tracking_groups:
-                        uplink["link_tracking_groups"].append({"name": lt_group["name"], "direction": "upstream"})
+    def _get_p2p_vrfs_uplinks(self: EosDesignsFacts) -> list:
+        """
+        List of uplinks with all parameters for uplink_type p2p-vrfs
+        """
+        uplinks = []
+        uplink_switches = self.shared_utils.uplink_switches
+        uplink_switch_interfaces = self._uplink_switch_interfaces
```

Comment:
The first part of this is boilerplate. Maybe we should move that to the base function and call per link to these functions

Reply:
yeah let me investigate this